cukinia

```diff
@@ -363,16 +363,45 @@ _cukinia_process()
 {
 	local pname="$1"
 	local puser="$2"
-	local ret
+	local proc_name
+	local proc_uid
 
 	_cukinia_prepare "Checking process \"$pname\" as ${puser:-any user}"
-	ret=$(ps -a -o user,comm |
-			awk '($2 == "'$pname'") && ($1 ~ /^'${puser:-.*}'$/) {
-			    print;
-			}')
-	[ -n "$ret" ] && return 0 || return 1
-}
 
+	if [ ! -z $puser ]; then
+		puser_id=`id -u $puser &>/dev/null`
```

Comment:
1. `&>` is not a **POSIX** shell valid redirection. I guess you want to redirect the *error* output so you want `2>/dev/null`.

2. could you double quote `$puser` (even if it is not necessary since your already check for `-n` earlier).

3. I would prefer to name the variable `puser_uid` to make the userid/groupid distinction.